project_issue_task_sync/models/project_issue.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, models
+
+
+class ProjectIssue(models.Model):
+    _inherit = 'project.issue'
+
+    def get_changed_vals(self, task):
+        vals = {}
+        if task.stage_id != self.stage_id:
+            vals['stage_id'] = self.stage_id.id
+        if task.user_id != self.user_id:
+            vals['user_id'] = self.user_id.id
+        return vals
+
+    def set_issue_vals(self):
+        for this in self:
+            task = this.sudo().task_id[:1]
+            # TODO task creation? for now if there is no task just skip.
+            if (this.project_id.sync_tasks_issues and task and not
+                    self.env.context.get('is_sync_operation')):
+                vals = this.get_changed_vals(task)
+                if vals:
+                    this.task.with_context(
+                        mail_notrack=True, is_sync_operation=True
+                    ).write(vals)
```

Comment:
When different, the values on the Issue overwrite the values on the Task.
This is not obvious, and can even be seen as unexpected:
In the worklflow where an Issue is reported, and then a Task is created for it's resolution, we would expect the Task stage and Responsible to change, and have the Issue sync with it.
AFAICS, with this code, if I try to change the Task Stage, the change would be automatically reverted!